--- /usr/share/lipstick-jolla-home-qt5/lockscreen/Clock.qml	2017-04-06 20:33:06.459835610 +0200
+++ /usr/share/lipstick-jolla-home-qt5/lockscreen/Clock.qml	2017-04-06 20:34:51.444842144 +0200
@@ -8,6 +8,7 @@
 import QtQuick 2.0
 import Sailfish.Silica 1.0
 import org.nemomobile.time 1.0
+import "../eventsview/weather"
 import org.nemomobile.lipstick 0.1
 import "../main"
 
@@ -85,4 +86,16 @@
         }
     }
 
+	WeatherLoader {
+        id: weatherWidget
+		visible: true
+		active: true
+		enabled: true
+	    anchors {
+			top: weekday.bottom
+			horizontalCenter: parent.horizontalCenter
+		}	
+		width: lockScreen.width
+    }		
+
 }
