--- /usr/share/lipstick-jolla-home-qt5/lockscreen/Clock.qml	2017-04-04 03:07:23.504361823 +0200
+++ /usr/share/lipstick-jolla-home-qt5/lockscreen/Clock.qml	2017-04-04 03:37:20.871291317 +0200
@@ -32,14 +32,39 @@
             bottomMargin: -timeText.font.pixelSize
             horizontalCenter: parent.horizontalCenter
         }
-        font { pixelSize: Theme.fontSizeHuge * 2.0; family: Theme.fontFamilyHeading }
+        font { pixelSize: Theme.fontSizeHuge * 3.5; family: Theme.fontFamilyHeading }
+        text: Qt.formatTime(new Date(), "hh")
     }
+    ClockItem {
+        id: timeTextMin
+        color: Theme.primaryColor
+        // Ascender of the time to the top of the clock.
+        anchors {
+            top: timeText.baseline
+            topMargin: -Theme.paddingLarge
+            bottom: parent.top
+            bottomMargin: -timeText.font.pixelSize
+            horizontalCenter: parent.horizontalCenter
+        }
+        font { pixelSize: Theme.fontSizeHuge * 3.5; family: Theme.fontFamilyHeading }
+        text: Qt.formatTime(new Date(), "mm")
+    }
+
+
+    Timer {
+        id: timerClock
+        interval: 60; running: true; repeat: true
+	onTriggered: {
+        	timeText.text = Qt.formatTime(new Date(), "hh")
+        	timeTextMin.text = Qt.formatTime(new Date(), "mm")   
+    	}
+     }
 
     Text {
         id: weekday
         anchors {
             horizontalCenter: parent.horizontalCenter
-            top: timeText.baseline
+            top: timeTextMin.baseline
             topMargin: Theme.paddingMedium
         }
         color: timeText.color
